web/src/timerender.ts

```diff
@@ -455,3 +455,27 @@ export function get_full_datetime(time: Date): string {
 
     return $t({defaultMessage: "{date} at {time}"}, {date: date_string, time: time_string});
 }
+
+export function get_time_limit_setting_in_appropriate_unit(time_limit_in_seconds: number): string {
+    const time_limit_in_minutes = Math.floor(time_limit_in_seconds / 60);
+    if (time_limit_in_minutes < 60) {
+        if (time_limit_in_minutes === 1) {
+            return time_limit_in_minutes.toString() + " minute";
+        }
+        return time_limit_in_minutes.toString() + " minutes";
+    }
+
+    const time_limit_in_hours = Math.floor(time_limit_in_minutes / 60);
+    if (time_limit_in_hours < 24) {
+        if (time_limit_in_hours === 1) {
+            return time_limit_in_hours.toString() + " hour";
+        }
+        return time_limit_in_hours.toString() + " hours";
+    }
+
+    const time_limit_in_days = Math.floor(time_limit_in_hours / 24);
+    if (time_limit_in_days === 1) {
+        return time_limit_in_days.toString() + " day";
+    }
+    return time_limit_in_days.toString() + " days";
+}
```

Comment:
This doesn't work from an i18n standpoint -- we're now combining English strings with numbers.

You should be able to fix it by just passing the time values into a translated string as the variable for the number, with ICU Plurals for the day/days part.